packages/mui-material/src/Button/Button.js

```diff
@@ -268,14 +281,43 @@ const ButtonStartIcon = styled('span', {
 
     return [styles.startIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
   },
-})(({ ownerState }) => ({
+})(() => ({
   display: 'inherit',
   marginRight: 8,
   marginLeft: -4,
-  ...(ownerState.size === 'small' && {
-    marginLeft: -2,
-  }),
-  ...commonIconStyles(ownerState),
+  variants: [
+    {
+      props({ ownerState }) {
+        return ownerState.size === 'small';
+      },
```

Comment:
```suggestion
      props: { size: 'small' }
```

Should work, right?

The same comment for the similar prop callbacks below

Reply:
Yes, thanks.